python/lib/modeldata/dmod/modeldata/data/item_domain_detector.py

```diff
@@ -0,0 +1,211 @@
+            return f"{matches.group(1)}-{matches.group(2)}"
+        else:
+            raise DmodRuntimeError(f"{self.__class__.__name__} couldn't parse cat id from name '{self._item_name}'")
+
+    def _get_cat_restriction(self) -> DiscreteRestriction:
+        """ Get :class:`DiscreteRestriction` defining applicable catchments (i.e., catchment) for the domain. """
+        return DiscreteRestriction(variable=StandardDatasetIndex.CATCHMENT_ID, values=[self._get_catchment_id()])
+
+    def detect(self, **kwargs) -> DataDomain:
+        """
+        Detect and return the data domain.
+
+        Parameters
+        ----------
+        kwargs
+            Optional kwargs applicable to the subtype, which may enhance or add to the domain detection and generation
+            capabilities, but which should not be required to produce a valid domain.
+
+        Returns
+        -------
+        DataDomain
+            The detected domain.
+
+        Raises
+        ------
+        DmodRuntimeError
+            If it was not possible to properly detect the domain.
+        """
+
+        # Do this early to fail here rather than try to load the dataframe
+        cat_restriction = self._get_cat_restriction()
+        data = StringIO(self._item.decode(self._decode_format)) if isinstance(self._item, bytes) else self._item
+        dt_index = self.get_data_format().indices_to_fields()[StandardDatasetIndex.TIME]
+        # TODO: (later) perhaps do a little more about the header checking
+        df = pandas_read_csv(data, parse_dates=[0])
+        self._num_time_steps = df.shape[0]
+        date_range = TimeRange(begin=df.iloc[0][dt_index].to_pydatetime(), end=df.iloc[-1][dt_index].to_pydatetime())
+        return DataDomain(data_format=self.get_data_format(), continuous_restrictions=[date_range],
+                          discrete_restrictions=[cat_restriction])
+
+
+# TODO: track and record hydrofabric ids and file names of what's out there so that we recognize known versions/regions
+
+
+# TODO: might need to expand things in the future ... there are other geopackage formats (e.g., older NextGen
+#  hydrofabric versions that used "id" instead of "divide_id") and maybe we need to account for that in detectors (and
+#  in formats)
+class GeoPackageHydrofabricDomainDetector(ItemDataDomainDetector):
+
+    _data_format = DataFormat.NGEN_GEOPACKAGE_HYDROFABRIC_V2
+
+    def _is_region_string_for_conus(self, region_str: Optional[str]) -> bool:
+        """
+        Whether this is a region string signifies CONUS.
+
+        Parameters
+        ----------
+        region_str: Optional[str]
+            A region string, or ``None``.
+
+        Returns
+        -------
+        bool
+            Whether there was a region string provided that indicates CONUS.
+        """
+        if not isinstance(region_str, str):
+            return False
+        else:
+            return region_str.strip().lower() == 'conus'
+
+    def detect(self, **kwargs) -> DataDomain:
```

Comment:
To maintain the "same" signature across different subtypes, but also allow for different optional behavior.  Which is probably a bad idea.  I'd like to leave it in for now though, rather than forget something necessary for properly removing it and replacing it with more appropriate implementation where needed.  Especially given current time constraints.

Reply:
That makes sense. As a compromise moving forward, in situations where we _need_ to accept `kwargs` but we don't use any of the `kwargs` in the implementation, I suggest that we use `**_` instead of `**kwargs`. This will indicate to the user and IDE / LSP that any keyword arguments are unused, but providing them won't break things.